pyrsistent/_helpers.py

```diff
@@ -27,12 +36,12 @@ def freeze(o):
     (1, pvector([]))
     """
     typ = type(o)
-    if typ is dict:
-        return pmap(dict((k, freeze(v)) for k, v in o.items()))
-    if typ is list:
-        return pvector(map(freeze, o))
+    if typ is dict or (not nonstrict and typ is pmap_type):
+        return pmap(dict((k, freeze(v, nonstrict)) for k, v in o.items()))
```

Comment:
If you want to these functions could probably be modernised a bit while you're at it. They were initially written with Python 2.6 compatibility in mind, hence the use of dict() for example. Also the `map` calls could perhaps be replaced with list comprehensions to allow skipping the lambdas as well. Not sure if a iterator expression (without square brackets) or  list comprehension is best from a perf standpoint. If you want to you could do some quick measurements.

If you don't feel like doing the above that's fine too. I can fix it at a later stage.

Reply:
Happy to do that. I'll have another look at it at some point this week and update thaw as well. I had some reasoning for using nonstrict instead of strict but tbh it is prob better just to use strict=False as the flag :)